cunumeric/sphinxext/comparison_table.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import annotations
+
+from dataclasses import dataclass
+from types import ModuleType
+
+from docutils.parsers.rst.directives import choice
+from sphinx.util.logging import getLogger
+
+from ..coverage import is_implemented, is_multi, is_single
+from ._comparison_config import (
+    GROUPED_CONFIGS,
+    MISSING_NP_REFS,
+    NUMPY_CONFIGS,
+    SKIP,
+)
+from ._cunumeric_directive import CunumericDirective
+from ._templates import COMPARISON_TABLE
+
+log = getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ItemDetail:
+    name: str
+    implemented: bool
+    np_ref: str
+    lg_ref: str
+    single: str
+    multi: str
+
+
+@dataclass(frozen=True)
+class SectionDetail:
+    title: str
+    np_count: int
+    lg_count: int
+    items: list[ItemDetail]
+
+
+def _npref(name, obj):
+    if isinstance(obj, ModuleType):
+        full_name = f"{obj.__name__}.{name}"
+    else:
+        full_name = f"numpy.{obj.__name__}.{name}"
+
+    role = "meth" if "ndarray" in full_name else "obj"
+
+    if full_name in MISSING_NP_REFS:
+        return f"``{full_name}``"
+    return f":{role}:`{full_name}`"
+
+
+def _lgref(name, obj, implemented):
+    if not implemented:
+        return "-"
+
+    if isinstance(obj, ModuleType):
+        full_name = f"{obj.__name__}.{name}"
+    else:
+        full_name = f"cunumeric.{obj.__name__}.{name}"
+
+    role = "meth" if "ndarray" in full_name else "obj"
```

Comment:
This is a bit clunky. If we ever need to document other classes, this will have to be generalized.